sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -62,6 +64,11 @@ object DateTimeUtils {
 
   @transient lazy val defaultTimeZone = TimeZone.getDefault
 
+  // These are ISO 8601 compliant `FastDateFormat`s. These are thread-safe and compatible
+  // with `SimpleDateFormat`.
+  lazy val iso8601TimeFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
```

Comment:
can you explain why this is a lazy val? (i see defaultTimeZone was also a lazy val but not sure why)